js/src/iaa.js

```diff
@@ -58,6 +58,39 @@ class IntervalAgreementApproach {
     );
     return top / bottom;
   }
+
+  /**
+   * Calculating mean of maxima is similar to centroid in that it will need to 
+   * sample linearly spaced values across a range (or ranges!), but then it will average all sampled values
+   */
+  get mean_of_maxima() {
+    const smallestValue = Math.min(...this.intervals.singletonKeys);
+    const largestValue = Math.max(...this.intervals.singletonKeys);
+    const values = linspace(
+      [smallestValue, largestValue],
+      (largestValue - smallestValue) * 10 + 1
+    )
+    const sum = values.reduce((a, b) => a + b, 0)
+    mean = sum/values.length
+    
+    return mean;
+  }
+
+  /**
+   * sum all minimum values and all maximum values then divide 
+   * by the count of all summed values (i.e. sum of the counts of min and max)
+   */
+  get mean_of_midpoints() {
+    const minValues = this.intervals.singletonKeys.filter(val=> val == Math.min(...this.intervals.singletonKeys))
```

Comment:
This only selects values which equal the min and max value of the whole dataset, but it needs all minimum values and maximum values from each individual interval.

So for the interval set `[1,5], [1,3], [2,5]`, the min values are `1,1,2` not just `1`, and the max values are `5,3,5` not just `5`.